### ISSUE TITLE
Question about the results on "Wander" dataset in Supp

### ISSUE THREAD

**Issue by lixiangyu-1008** [nianticlabs/monodepth2#227]
Hi @mrharicot @amonszpart @macaodha @mdfirman  Thanks for your awesome works.
I have a question on the training of the  "Wander" dataset as described in Supplementary material.
As you download the dataset from Youtube, so what the intrinsic as you used in this training?

**Comment by mrharicot** [comment]
Hi,
Thank you for the kind words!
Here's the focal length that we used for training. 
```python
self.K = np.array([[0.6, 0, 0.5, 0],
                   [0, 1.066, 0.5, 0],
                   [0, 0, 1, 0],
                   [0, 0, 0, 1]], dtype=np.float32)
```
It was originally obtained when I ran colmap on one of the sequences, however it's likely that the focal might be different for each sequence.

I hope this helps!

**Comment by lixiangyu-1008** [comment]
hi, Thanks for patient reply.
And for the **feature reconstruction loss in the appearance matching term**， can you share more details or some implemented repo :)

**Comment by mrharicot** [comment]
Here's the code I originally wrote to get the first conv out of vgg16.
This wasn't optimized or thoroughly tested and I am only providing this as an example.
```python
class VGGFeatures(nn.Module):
    def __init__(self, normalize_input=True):
        super(VGGFeatures, self).__init__()
        self.model = torchvision.models.vgg16(True)
        self.conv = list(self.model.features.children())[0]
        self.conv.padding = 0
        self.relu = nn.ReLU(True)

        self.refl = nn.ReflectionPad2d(1)

        self.normalize_input = normalize_input

        for param in self.model.parameters():
            param.requires_grad = False

    def forward(self, x):
        if self.normalize_input:
            x = (x - 0.45) / 0.225
        out = self.refl(x)
        out = self.conv(out)
        out = self.relu(out)
        out = F.normalize(out)
        return out
```

The loss then becomes something like:
```python
vgg_features = VGGFeatures().cuda()

...

inputs[("feat", frame_id, source_scale)] = vgg_features(inputs[("color", frame_id, source_scale)])

...

outputs[("feat", frame_id, scale)] = F.grid_sample(
                    inputs[("feat", frame_id, source_scale)],
                    outputs[("sample", frame_id, scale)],
                    padding_mode="border")

loss = torch.abs(inputs[("feat", 0, source_scale)] - outputs[("feat", frame_id, scale)]).mean()
```

I hope this helps!

**Comment by lixiangyu-1008** [comment]
@mrharicot , Thanks a lot. :)
